ddtrace/span.py

```diff
@@ -320,7 +321,12 @@ def _set_str_tag(self, key, value):
         str in Python 3, with decoding errors in conversion being replaced with
         U+FFFD.
         """
-        self.meta[key] = ensure_text(value, errors="replace")
+        try:
+            self.meta[key] = ensure_text(value, errors="replace")
+        except Exception as e:
+            if config._raise:
+                raise e
+            log.warning("Failed to set text tag '%s'", key, exc_info=True)
```

Comment:
should we add the span to this log as well? maybe just the operation name?

trying to think if this warning is valuable enough on it's own

the original value too maybe?

Reply:
The traceback should tell us about the original value. Also, not sure what knowing about the span would help us debug. Any specific situation you were thinking, @brettlangdon ?

Reply:
@brettlangdon the traceback would probably include some additional details (about the original value, the integration it's coming from etc...) and it's probably noisy enough. Do you have some particular use of span data in mind?

Reply:
no, you are probably right that the traceback is probably enough to see the source of who is setting it

I was mostly thinking for a customer if it said "Failed to set text tag 'django.view' from None" or something, it might be much easier to debug at a glance

but again, traceback is probably fine